handlers/validator.go

```diff
@@ -1897,6 +1898,17 @@ func ValidatorStatsTable(w http.ResponseWriter, r *http.Request) {
 
 	errFields["index"] = index
 
+	// Check if validator index is available
+	var dbIndex uint64
+	err = db.ReaderDb.Get(&dbIndex, "SELECT validatorindex FROM validators WHERE validatorindex = $1", index)
+	if err != nil {
+		if err != sql.ErrNoRows {
+			utils.LogError(err, "error checking for index in validators table", 0, errFields)
+		}
+		validatorNotFound(data, w, r, vars, "/stats")
+		return
+	}
```

Comment:
**suggestion:** Since we only want to check whether the index exists we can use the EXISTS keyword.
```suggestion
	var doesIndexExist bool
	err = db.ReaderDb.Get(&doesIndexExist, "SELECT EXISTS (SELECT validatorindex FROM validators WHERE validatorindex = $1)", index)
	if err != nil || !doesIndexExist {
		if err != nil {
			utils.LogError(err, "error checking for index in validators table", 0, errFields)
		}
		validatorNotFound(data, w, r, vars, "/stats")
		return
	}
```